components/save_button.jsx

```diff
@@ -20,14 +20,20 @@ export default class SaveButton extends React.PureComponent {
 
     static defaultProps = {
         disabled: false,
-        savingMessage: localizeMessage('save_button.saving', 'Saving'),
-        defaultMessage: localizeMessage('save_button.save', 'Save'),
         btnClass: 'btn-primary',
         extraClasses: '',
     }
 
     render() {
-        const {saving, disabled, savingMessage, defaultMessage, btnClass, extraClasses, ...props} = this.props; // eslint-disable-line no-use-before-define
+        const {
+            saving,
+            disabled,
+            savingMessage = localizeMessage('save_button.saving', 'Saving'),
+            defaultMessage = localizeMessage('save_button.save', 'Save'),
```

Comment:
Can we have these set in `defaultProps`?

Non-blocking but just like to mention that there's an existing [tech debt ticket](https://mattermost.atlassian.net/browse/MM-13303) that aimed to remove `Utils.localizeMessage`.

Reply:
@saturninoabril, I can't put these back into `defaultProps`: that's precisely what broke the issue in the first place. The default props are never re-evaluated, so the english translations remains in place until the page refreshes.

Agree that solving the tech debt here would be best.

Reply:
Should I just kick these over to `FormattedMessage` instances instead?

Reply:
Oh sorry for that, it's me not paying attention to the change :)

Yes or `formatMessage` from `this.context.intl` if you choose to pass it via context.

Reply:
No worries! I've switched to `formatMessage`, since we require the actual text for one of the properties being passed downstream.